### PR TITLE
Add related video and MPU

### PR DESCRIPTION
**Pull request by ironsidevsquincy**
screen shot 2015-06-10 at 11 37 46

### CODE REVIEW

**Comment by matthew-andrews** [comment]
woww!!

File: views/partials/article-list.html [hunk 1]
```diff
@@ -1,15 +1,32 @@
...
+		{{#slice items limit="7"}}
+			<li class="stream-item stream-item--row o-grid-row">
+				{{#if template}}
+					{{{usePartial template}}}
+				{{else}}
+					{{>next-card/templates/card}}
+				{{/if}}
+			</li>
+		{{/slice}}
+		{{#if showRelatedContent}}
```
Comment (matthew-andrews) [inline views/partials/article-list.html]: hmm seems like we have two solutions here…… could we not just put the related content thing in the stream array itself with the  `template` set to not an article card?

Comment (matthew-andrews) [inline views/partials/article-list.html]: [instead of slicing in the view]

Comment (ironsidevsquincy) [inline views/partials/article-list.html]: yep, good point

File: server/controllers/v1-taxonomy.js [hunk 2]
```diff
@@ -62,7 +62,10 @@ module.exports = function(req, res, next) {
 				layout: 'article-list',
 				symbol: symbol,
 				currentNav: currentNav && currentNav.children ? currentNav : false,
-				taxonomy: taxonomy
+				taxonomy: taxonomy,
+				// only show on top level sections for now
+				showRelatedContent: res.locals.flags.relatedContentOnStreams &&
```
Comment (matthew-andrews) [inline server/controllers/v1-taxonomy.js]: also should we limit this to just be on page 1 of streams?

Comment (ironsidevsquincy) [inline server/controllers/v1-taxonomy.js]: hmmm...for now, yes

File: client/components/stream/related.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+		if (!relatedItemsContainerEl) {
+			return;
+		}
+		var tagName = relatedItemsContainerEl.getAttribute('data-tag-name');
+		var searchTagName = tagName === 'Life & Arts' ? 'Life and Arts' : tagName;
+		var searchQuery = 'all=tag:section:' + searchTagName;
+
+		fetch('//next-brightcove-proxy.ft.com/search?' + searchQuery + '&page_size=2')
+			.then(function (response) {
+				return response.json();
```
Comment (matthew-andrews) [inline client/components/stream/related.js]: what's the behaviour if this response is not `.ok`?

File: client/components/stream/related.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+			})
+			.then(function (json) {
+				if (json.items.length === 0) {
+					return;
+				}
+				var relatedItemsTitleEl = document.createElement('h2');
+				relatedItemsTitleEl.className = 'related-items-title';
+				relatedItemsTitleEl.innerHTML = '<span class="related-items-title__tag-name">' + tagName +
```
Comment (matthew-andrews) [inline client/components/stream/related.js]: es6 template string might be neater here, if you like :-)

Comment (matthew-andrews) [inline client/components/stream/related.js]: [very very optional]

Comment (ironsidevsquincy) [inline client/components/stream/related.js]: Amazing. Is that browserify doing the magic?

Comment (matthew-andrews) [inline client/components/stream/related.js]: OBT :wink:

Comment (ironsidevsquincy) [inline client/components/stream/related.js]: ahh, uses babelify, need to make more use of this

Comment (matthew-andrews) [inline client/components/stream/related.js]: client side only atm

**Pull request merged**